### PR TITLE
Fix tenant and reconcile commands

### PR DESCRIPTION
**Pull request by stefanprodan**
Changes:
- Add service account namespace to tenant role binding
- Rename role binding to match the tenant name
- Do not try to reconcile a suspended object

### CODE REVIEW

File: cmd/flux/create_tenant.go [hunk 1]
```diff
@@ -123,18 +122,20 @@ func createTenantCmdRun(cmd *cobra.Command, args []string) error {
...
 				Namespace: ns,
 				Labels:    objLabels,
 			},
 			Subjects: []rbacv1.Subject{
 				{
-					Kind: "User",
-					Name: fmt.Sprintf("gotk:%s:reconciler", ns),
+					APIGroup: "rbac.authorization.k8s.io",
+					Kind:     "User",
+					Name:     fmt.Sprintf("gotk:%s:reconciler", ns),
```
Comment (phillebaba) [inline cmd/flux/create_tenant.go:133]: Can users have non dns-resolvable names?

Comment (stefanprodan) [reply cmd/flux/create_tenant.go:133]: no, see https://github.com/fluxcd/flux2/blob/17ba4464929a45ae7f90d622fdcb232d0f2c1e90/cmd/flux/create_tenant.go#L77

**Review by phillebaba — APPROVED** [review]
LGTM

**Review by hiddeco — COMMENTED** [review]
I would put the annotation change in a separate PR as it should be included in the next MINOR release.

**Comment by stefanprodan** [comment]
@hiddeco the annotation change was included in 0.3, should we wait for 0.5?

**Comment by hiddeco** [comment]
@stefanprodan the change itself can’t be included in a PATCH release as it is breaking, so yes.

**Comment by stefanprodan** [comment]
@hiddeco I removed the annotations from this PR.

**Review by hiddeco — APPROVED** [review]
:+1:

**Pull request merged**